### ISSUE TITLE
Enable GitHub PR Merge Queues for this project.

### ISSUE THREAD

**Issue by aloisklink** [nqminds/edgesec#411]
**Is your feature request related to a problem? Please describe.**

The `edgesec` GitHub repo enforces that all PRs must be up-to-date when merging a PR. However, this makes merging multiple PRs in succession quite tedious, as we need to manually update each PR, wait for CI to pass, then merge them.

**Describe the solution you'd like**

On 2023-02-08, GitHub announced that the PR Merge Queue beta is now public for any open-source repo to join, see https://github.blog/changelog/2023-02-08-pull-request-merge-queue-public-beta/

![image](https://user-images.githubusercontent.com/19716675/218061474-8ff2fbe2-e97d-4107-81f9-64f349e647c8.png)

Basically, when we do try to merge, it will automatically try multiple PRs at once, so it should speed up merging by quite a bit. If a PR has a failed **required** check, it should automatically be pulled from the queue.

**Describe alternatives you've considered**

N/A

**Additional context**

We'd probably also want to make a bunch more CI checks **required**, e.g. pretty-much everything except the code-coverage and **Code scanning results / CodeQL** should be **required** IMO.

Any thoughts?

**Comment by mereacre** [comment]
I think we should enable it and add more tests on the way.

**Comment by aloisklink** [comment]
Seems to work well!

https://github.com/nqminds/edgesec/pull/412 merged fine :)

We can re-open this issue if we have any problems with it.
